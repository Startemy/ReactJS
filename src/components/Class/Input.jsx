import React, { Component } from "react";

export class Input extends Component {
  state = {
    value: ''
  }

  render() {
    return <>
    <input type="text" value={ this.props.value } 
      onChange={ this.props.change }/>
    </>
    
  }
}