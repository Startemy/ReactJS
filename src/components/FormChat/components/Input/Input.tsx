import React, { FC } from 'react';

interface NameChatProps {
  value: string,
  change: React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement>,
}

export const Input: FC<NameChatProps> = ({ value, change }) => (
  <input
    name='nameChart'
    value={value}
    onChange={change}
    placeholder="Введите название чата"
  />
);