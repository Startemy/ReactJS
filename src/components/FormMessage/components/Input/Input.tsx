import React, { forwardRef } from 'react';

interface InputProps {
  value: string;
  change: React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement>;
}

const Input = forwardRef<HTMLTextAreaElement, InputProps>(
  ({ value, change }, ref) => (
    <textarea
      ref={ref}
      name="msg"
      value={value}
      onChange={change}
      placeholder="Писать сюда"
    />
  )
);

Input.displayName = 'Input';
export { Input };
