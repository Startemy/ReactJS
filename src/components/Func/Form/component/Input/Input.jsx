import React, { forwardRef } from 'react';

export const Input = (props, ref) => {
  return (
    <textarea
      ref={ref}
      name="msg"
      value={props.value}
      onChange={props.change}
      onKeyDown={props.onkey}
      placeholder="Писать сюда"
    ></textarea>
  );
};

export default forwardRef(Input);
