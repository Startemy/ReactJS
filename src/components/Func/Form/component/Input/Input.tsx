import React, { forwardRef, FC } from 'react';
// import InputUI from '@mui/material/TextField';

interface InputProps {
  value: string,
  change: React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement>,
  onkey: React.KeyboardEventHandler<HTMLTextAreaElement | HTMLInputElement>
}

export const Input: FC<InputProps>  = (props, ref) => (
  <textarea
    ref={ref}
    name="msg"
    value={props.value}
    onChange={props.change}
    onKeyDown={props.onkey}
    placeholder="Писать сюда"
  ></textarea>

  /** mui посмотреть как привязать ref */
  // <InputUI
  //   ref={ref}
  //   id="standard-multiline-static"
  //   label="Ваше сообщение"
  //   multiline
  //   rows={2}
  //   variant="standard"
  //   name="msg"
  //   value={props.value}
  //   onChange={props.change}
  //   onKeyDown={props.onkey}
  //   placeholder="Писать сюда">
  // </InputUI>
);

export default forwardRef(Input);
