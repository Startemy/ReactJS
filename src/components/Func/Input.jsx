import React, { forwardRef } from 'react';

export const Input = (props, ref) => {
  return (
    <textarea
      ref={ref}
      type="text"
      value={props.value}
      onChange={props.change}
      onKeyDown={props.onkey}
      placeholder="Писать сюда"
    ></textarea>
  );
};

export default forwardRef(Input);
