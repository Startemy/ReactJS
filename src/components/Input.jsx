import React, { Component } from "react";

export class Input extends Component {
  state = {
    value: ''
  }

  hendlerInput = (event) => {
    this.setState({ value: event.target.value })
  }

  render() {
    return <>
    <p>{ this.state.value }</p>
    <input type="text"  value={ this.state.value } onChange={ this.hendlerInput }/>
    </>
    
  }
}