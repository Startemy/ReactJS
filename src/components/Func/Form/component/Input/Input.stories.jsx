import React from 'react';
import Input from './Input';

export default {
  title: 'myComp/Input',
  component: Input,
  argTypes: {
    handleChange: { action: 'change' },
    onKye: { action: 'onKey' },
  },
};

const Template = (args) => <Input {...args} />;

export const Firsst = Template.bind({});
Firsst.args = {
  value: '',
};
