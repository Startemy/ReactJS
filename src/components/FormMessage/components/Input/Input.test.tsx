import React from 'react';
import { Input } from './Input';
import { render, screen } from '@testing-library/react';
import 'jest-enzyme';
import '@testing-library/jest-dom';

const TestInput = (props: any) => <Input {...props} />;

describe('Input', () => {
  it('render component', () => {
    render(<TestInput />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<TestInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render with snapshot', () => {
    render(<TestInput />);
    expect(screen.getByPlaceholderText('Писать сюда')).toBeTruthy();
  });
});
