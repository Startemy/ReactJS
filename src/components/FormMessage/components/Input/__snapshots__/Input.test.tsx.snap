// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render with snapshot 1`] = `
<DocumentFragment>
  <textarea
    name="msg"
    placeholder="Писать сюда"
  />
</DocumentFragment>
`;
